pip_api/_parse_requirements.py

```diff
@@ -187,16 +187,21 @@ def parse_requirements(
             req: Optional[Union[requirements.Requirement, UnparsedRequirement]] = None
             known, _ = parser.parse_known_args(line.strip().split())
             if known.req:
+                # We want to stop combining strings before an @ sign with space on either side
```

Comment:
I realise this block of code is weird.

The tests in `test_parse_requirements.py` are very specific about how `package@url` should be treated vs `package @ url` so that's why I've written it like this instead of just doing a `str().join(known.req)`.